dotcom-rendering/src/model/enhance-placeholders.ts

```diff
@@ -0,0 +1,83 @@
+import { Design } from '@guardian/types';
+import { decideDesign } from '@root/src/web/lib/decideDesign';
+
+export const replacePlaceholders = (
+	elem: InteractiveAtomBlockElement,
+	variables: Map<string, string>,
+): InteractiveAtomBlockElement => {
+	if (!elem.html) {
+		return elem;
+	}
+
+	const replacer = (substring: string, name: string): string => {
+		return variables.get(name) || substring;
+	};
+
+	// Note: use str.replace with 'g' (global) flag as str.replaceAll not
+	// supported in current node version (14.6).
+	const re = /{{\s*(\w+)\s*}}/g;
+	elem.html = elem.html.replace(re, replacer);
+
+	return elem;
+};
+
+/**
+ * generateVariables defines the templated variables we replace. As such, it
+ * represents an ongoing contract with interactive atoms; fields should not be
+ * removed from it once added so be cautious when thinking about adding new
+ * ones.
+ */
+const buildPlaceholders = (data: CAPIType): Map<string, string> => {
+	return new Map(
+		Object.entries({
+			webPublicationDate: data.webPublicationDateDisplay || '',
+			byline: data.author.byline || '',
+			webTitle: data.webTitle,
+			trailText: data.trailText,
+			sectionLabel: data.sectionLabel,
+			// shareLinks: TODO,
```

Comment:
cc @HarryFischer anything else we could include to start here?

Reply:
standfirst?

Reply:
Why don't they all have fallbacks? Are some of them required elsewhere, in Composer?

Reply:
Also wondering if headline should exist here? I think there is sometimes a distinction between title and headline